### ISSUE TITLE
Default to Trusty

### ISSUE THREAD

**Issue by CorbanR** [hashicorp/otto#442]
Currently otto defaults to using precise for it's vagrant base.  It would be nice if it could be updated to trusty.

**Comment by leowmjw** [comment]
It was easy enough to just modify the layer-base Vagrantfile to use ubuntu/trusty64 instead of hashicorp/precis64.

Unfortunately; the created layers seems to be suffering from the problem observed at https://github.com/mitchellh/vagrant/issues/5186 (e.g unable to ssh in); so I had to revert back to hashicorp/precise64

So :+1:  for a hashicorp/trusty64 box

**Comment by legal90** [comment]
Probably, it makes sense to use bento box "bento/ubuntu-14.04" instead.
[Bento](https://github.com/chef/bento) is a community-driven project, managed by Chef.

**Comment by winglian** [comment]
Perhaps a better solution is to allow us to configure the vagrant base image rather than waiting for the maintainers to change it and release a new build whenever things like https://github.com/hashicorp/otto/issues/496 break or having to build it ourselves from source.

**Comment by winglian** [comment]
My workaround for now is to run `otto compile` and then edit `.otto/compiled/app/foundation-layers/0-consul/Vagrantfile` to use `bento/ubuntu-14.04` per @legal90; all before finally running `otto dev`
Looks like the proper line to edit for a PR to update this is https://github.com/hashicorp/otto/blob/master/helper/compile/data/app/dev/Vagrantfile-layer.tpl#L12

**Comment by joelnb** [comment]
This didn't fix the issue for me, even when I replaced the precise box in every Vagrantfile inside `.otto` (I tried just the one you said first)

**Comment by winglian** [comment]
@joelnb did you do an otto destroy first?

**Comment by joelnb** [comment]
@winglian I'm pretty sure I did but I will try again later to be absolutely certain I haven't done something silly. Also, does otto handle this properly when it does things like install docker? Or will it still try and install the precise version on trusty?

**Comment by joelnb** [comment]
Yeah I think that the issue is that `otto dev destroy` doesn't destroy all of the layers & they were already created from the hashicorp box. I copied my project to a fresh machine, ran `otto compile`, then ran:

    for f in $(find . -name Vagrantfile); do
        sed -i "s|hashicorp/precise64|bento/ubuntu-14.04|" $f
    done

Then when I ran `otto dev` everything worked fine. Possibly this is more than required or poissibly this is required for me because I have a different type of project (it's a rails project so it is installing redis using docker).

**Comment by matthewhatch** [comment]
@joelnb I think starting with a fresh machine was part of your solution.  I was in the same situation and had the same issue even after replacing `hashicorp/precise64` with `bento/ubuntu-14.04` in all otto generated files.  The reason for this is all the base boxes still exist that the development vm is cloned from. So I had to remove all file/folders from `~/.otto` and remove all VMs created (this is necessary, but helps cleanup unused VMs)

`rm -rf ~/.otto`
`otto compile`
replaced `hashicorp/precise64` with `bento/ubuntu-14.04` in all otto generated files
'otto dev`

**Comment by gnulnx** [comment]
I ended up trying to follow @matthewhatch  steps without success... 

In my project directory I did

`otto dev destroy`
`rm -rf .otto*`
`otto compile`

Replace all instances of `hashicorp/precise64` with `bento/ubuntu-14.04`

`otto dev`
`otto ssh`
`lsb_release -d`

which unfortunately returns  "Description:	Ubuntu 12.04 LTS"

As of yet I am unable to change the base box type.  I am going to dig into the source to figure this out as this is a game changer on being able to use it right now.  :(

**Comment by legal90** [comment]
@gnulnx I guess that the reason is in "layers". You should delete to create a new VM from the scratch:
```
otto dev destroy
otto dev layers -prune
otto dev
# ...
```

**Comment by joelnb** [comment]
@gnulnx I think you will have the issue I did - the base layers otto has created still exist with the old image. I think the solution is to run `vagrant global-status` and for every vagrant VM in `~/.otto.d/` you can `cd` into the directory and run `vagrant destroy -f`. This should destroy all of the base layers. Then when you do the steps you did before and it should work.

**Comment by legal90** [comment]
> I think the solution is to run vagrant global-status and for every vagrant VM in ~/.otto.d/ you can cd into the directory and run vagrant destroy -f

`otto dev layers -prune` actually does that :)

**Comment by gnulnx** [comment]
Thanks gang that seemed to have worked with a couple of minor details.  Here is the full list of instructions I used to get this working.

Inside the app directory.

`otto dev destroy`
`rm -rf .otto*`
`otto dev layers -prune`

the prune command failed for me with an error like

> There was an error while executing `VBoxManage`, a CLI used by Vagrant
> for controlling VirtualBox. The command and stderr is shown below.

> Command: ["unregistervm", "7de50809-5d67-4cac-8459-3233370b044b", "--delete"]

> Stderr: 0%...10%...20%...30%...40%...50%...
> Progress state: VBOX_E_INVALID_OBJECT_STATE
> VBoxManage: error: Machine delete failed
> VBoxManage: error: Medium '/Users/jfurr/VirtualBox VMs/python27_default_1462978679656_27464/Snapshots/{145caeb5-f059-452a-93f0-> 9dc039e4dc9b}.vmdk' is locked for reading by another task
> VBoxManage: error: Details: code VBOX_E_INVALID_OBJECT_STATE (0x80bb0007), component > Medium, interface IMedium
> VBoxManage: error: Context: "int handleUnregisterVM(HandlerArg*)" at line 166 of file > VBoxManageMisc.cpp

So I fired up VirtualBox and found a lot of running instances and a few stale instances.. I stopped the running instances and then deleted them and removed all files.  this can be done outside VirtualBox, but this is how I did it.  Afterwards I did:

`otto compile`

Replace all instances of `hashicorp/precise64` with `bento/ubuntu-14.04`

`otto dev`
`otto ssh`
`lsb_release -d`

which now correctly spits out:
Description:	Ubuntu 14.04.4 LTS

Gracias!

**Comment by dring1** [comment]
Using VMWare fusion I ran into many issues with  `bento/ubuntu-14.04` - used `puphpet/ubuntu1404-x64` instead.

**Comment by goruha** [comment]
+1

**Comment by joshRpowell** [comment]
@gnulnx thanks for the roadmap. fixed my issue.  

** can we define the box with an env variable? **

`config.vm.box = ENV["OTTO_VAGRANT_BOX"]` 

by passing in something like? 

`otto dev -box bento/ubuntu-14.04`

i see we have the following: 

`otto dev vagrant box` options
